config/main.py

```diff
@@ -1667,6 +1673,65 @@ def load_port_config(config_db, port_config_path):
                 port_name), display_cmd=True)
     return
 
+
+def load_golden_config(golden_config_path):
+    if not os.path.isfile(golden_config_path):
+        return
+
+    # Override configDB with golden config
+    clicommon.run_command('config override-config-table {}'.format(
+        golden_config_path), display_cmd=True)
+    return
+
+
+#
+# 'override-config-table' command ('config override-config-table ...')
+#
+@config.command('override-config-table')
+@click.argument('golden-config-db', required=True)
+@click.option(
+    '--dry_run', type=click.STRING,
+    help="Dry run, writes config to the given file"
+)
+@clicommon.pass_db
+def override_config_table(db, golden_config_db, dry_run):
+    """Override current configDB with golden config."""
+
+    try:
+        # Load golden config json
+        golden_config_input = read_json_file(golden_config_db)
+    except Exception:
+        click.secho("Bad format: json file broken", fg='magenta')
+        sys.exit(1)
+
+    # Validate if the input is dict
+    if not isinstance(golden_config_input, dict):
+        click.secho("Bad format: golden_config_db is not a dict", fg='magenta')
+        sys.exit(1)
+
+    config_db = db.cfgdb
+
+    if dry_run:
+        # Read config from configDB
+        current_config = config_db.get_config()
+        # Serialize to the same format as json input
+        sonic_cfggen.FormatConverter.to_serialized(current_config)
+        # Override current config with golden config
+        for table in golden_config_input:
+            current_config[table] = golden_config_input[table]
+        with open(dry_run, "w+") as f:
+            json.dump(current_config, f, sort_keys=True, indent=4, cls=minigraph_encoder)
+    else:
+        # Deserialized golden config to DB recognized format
+        sonic_cfggen.FormatConverter.to_deserialized(golden_config_input)
+        # Delete table from DB then mod_config to apply golden config
+        click.echo("Removing configDB overriden table first ...")
+        for table in golden_config_input:
+            config_db.delete_table(table)
+        click.echo("Writing golden config to configDB ...")
+        data = sonic_cfggen.FormatConverter.output_to_db(golden_config_input)
+        config_db.mod_config(data)
```

Comment:
>mod_config [](http://example.com/codeflow?start=18&length=10)

If user invoke this CLI directly, is it possible to warn user that no services are restarted.